src/commands/reminder/CreateReminder.ts

```diff
@@ -0,0 +1,170 @@
+
+        // checks for [duration] if there's any button presses
+        // if no interaction within [duration], just exit
+        while (true) {
+            const interact = await AdvancedCollector.startInteractionEphemeralCollector(
+                {
+                    acknowledgeImmediately: false,
+                    duration: 2 * 60 * 1000,
+                    targetAuthor: ctx.user,
+                    targetChannel: ctx.channel,
+                    clearInteractionsAfterComplete: true
+                },
+                uniqueId
+            );
+            
+            // if there's no button press within [duration], delete it
+            if (!interact) {
+                break;
+            }
+
+            if (interact.isButton()) {
+                if (interact.customId === `${uniqueId}_date`) {
+                    AdvancedCollector.sendTextModal(interact, 
+                        {
+                            modalTitle: "Set Date (24 Hour Time)",
+                            inputs: [
+                                new TextInputBuilder()
+                                    .setStyle(TextInputStyle.Short)
+                                    .setMaxLength(2)
+                                    .setLabel("Month (1-12)")
+                                    .setCustomId("month")
+                                    .setRequired(true),
+                                new TextInputBuilder()
+                                    .setStyle(TextInputStyle.Short)
+                                    .setMaxLength(2)
+                                    .setLabel("Day (1-31)")
+                                    .setCustomId("day")
+                                    .setRequired(true),
+                                new TextInputBuilder()
+                                    .setStyle(TextInputStyle.Short)
+                                    .setMaxLength(4)
+                                    .setLabel("Year (xxxx)")
+                                    .setCustomId("year")
+                                    .setRequired(true),
+                                new TextInputBuilder()
+                                    .setStyle(TextInputStyle.Short)
+                                    .setMaxLength(2)
+                                    .setLabel("Hour (1-24)")
+                                    .setCustomId("hour")
+                                    .setRequired(true),
+                                new TextInputBuilder()
+                                    .setStyle(TextInputStyle.Short)
+                                    .setMaxLength(2)
+                                    .setLabel("Minute (1-60)")
+                                    .setCustomId("minute")
+                                    .setRequired(true),
+                            ],
+                            duration: 60 * 1000
+                        }, async (result) => {
+                            await result.deferUpdate();
```

Comment:
I don't think you need to `deferUpdate` here -- the operations you're doing won't take more than 5 seconds.